src/chart/speclayout/tablechart.h

```diff
@@ -18,14 +18,25 @@ template <typename Item>
 void TableChart::setupVector(std::vector<Item> &items,
     bool singleColumn)
 {
-	if (items.empty()) return;
-
 	auto size = 0;
 	for (auto &item : items)
 		if (item.enabled) ++size;
 
-	auto rowsize =
-	    singleColumn ? 1 : static_cast<ssize_t>(ceil(sqrt(size)));
+	if (size == 0) return;
+
+	ssize_t rowsize{1};
+	if (!singleColumn) {
+		rowsize = static_cast<ssize_t>(ceil(sqrt(size)));
+		auto sq = static_cast<ssize_t>(ceil(sqrt(
+		    16.0 * static_cast<double>(rowsize * rowsize) / 9.0)));
+
+		auto rem = rowsize - 1 - (size - 1) % rowsize;
```

Comment:
Good idea, but it will cause strange rearrangements in case the user filters out elements from the chart. 
If I understand it correctly, I have a 7x11 element in 7 rows, if I filter one out, it will rearrange it to 9 rows.

Reply:
The same happens in the origin version when it filters out enough markers to decrease the active markers count square root, or the filtered markers count does not match the sqrt of valid marker size. The rearrangement is always strange when dimension sizes are different than the sqrt (just think about 4 * 11).

This code at least tries to minimize the "empty" spaces.

Reply:
The change is bigger with the new version, on the other hand it was not a probllem imho, that it did not fill the fill the space, we are just listing the markers here, it's like word wrap for a text paragraph, the last line won't go till the end of line.

Let's ask for a second opinion. @andraskangyal, please read the discussion, then check out the difference:
Old: https://jsfiddle.net/simzer/um6jh5cv/6/
new: https://jsfiddle.net/simzer/um6jh5cv/7/

Reply:
If I'm looking at the 'Old' one, I feel it's better because the line shifting animation is clearer. But I don't know is there any other consequences too behind the new concept.(?)

Reply:
New:
pro: we have full last line more times, and we don't have only few or one markers in the last line.
contra: the row count goes up-down-up-down while we only adding the markers.

Reply:
Andris's decision: we go with the new one. but 16:9 should be changed to 1:1